docs/src/rules/no-restricted-imports.md

```diff
@@ -237,137 +202,147 @@ import * as Foo from "foo";
+/*eslint no-restricted-imports: ["error", { paths: [{
+    name: "foo",
+    importNames: ["DisallowedObject"],
+    message: "Please import 'DisallowedObject' from '/bar/baz/' instead."
 }]}]*/
 
-import pick from 'fooBar';
+import { AllowedObject as DisallowedObject } from "foo";
 ```
 
 :::
 
-::: incorrect { "sourceType": "module" }
+### patterns
 
-```js
-/*eslint no-restricted-imports: ["error", { patterns: [{
-    group: ["utils/*"],
-    importNames: ['isEmpty'],
-    message: "Use 'isEmpty' from lodash instead."
-}]}]*/
+This is also an object option whose value is an array. This option comes handy when you need to restrict the modules present inside a perticular directory. Simply, you can specify the modules inside the directories you want to restrict as string in `gitignore-style` patterns.
 
-import { isEmpty } from 'utils/collection-utils';
+Since the patterns follows the `gitignore-style` then if you want to reinclude any particular module this can be done by prefixing a negation (`!`) mark in front of the pattern and it's good to keep the negation patterns in last as `gitignore-style` follows order:
+
+```json
+"no-restricted-imports": ["error", {
+    "patterns": ["import1/private/*", "import2/*", "!import2/good"]
+}]
 ```
 
-:::
+Examples of **incorrect** code for `pattern` option:
 
 ::: incorrect { "sourceType": "module" }
 
 ```js
-/*eslint no-restricted-imports: ["error", { patterns: [{
-    group: ["utils/*"],
-    importNamePattern: '^is',
-    message: "Use 'is*' functions from lodash instead."
-}]}]*/
+/*eslint no-restricted-imports: ["error", { "patterns": ["lodash/*"] }]*/
 
-import { isEmpty } from 'utils/collection-utils';
+import pick from 'lodash/pick';
 ```
 
 :::
 
 ::: incorrect { "sourceType": "module" }
 
 ```js
-/*eslint no-restricted-imports: ["error", { patterns: [{
-    group: ["foo/*"],
-    importNamePattern: '^(is|has)',
-    message: "Use 'is*' and 'has*' functions from baz/bar instead"
-}]}]*/
+/*eslint no-restricted-imports: ["error", { "patterns": ["lodash/*", "!lodash/pick"] }]*/
 
-import { isSomething, hasSomething } from 'foo/bar';
+import pick from 'lodash/map';
 ```
 
 :::
 
 ::: incorrect { "sourceType": "module" }
 
 ```js
-/*eslint no-restricted-imports: ["error", { patterns: [{
-    group: ["foo/*"],
-    importNames: ["bar"],
-    importNamePattern: '^baz',
-}]}]*/
+/*eslint no-restricted-imports: ["error", { "patterns": ["import1/*", "!import1/private/*"] }]*/
 
-import { bar, bazQux } from 'foo/quux';
+import pick from 'import1/private/someModule';
 ```
 
 :::
 
-Examples of **correct** code for this rule:
+In last example `"!import1/private/*"` is not reincluding the modules inside private directory because in `gitignore-style` negation mark (`!`) does not reinclude the files if it's parent directory is excluded by a pattern. In this case `import1/private` directory is already excluded by the `import1/*` pattern, but excluded directory can be reincluded like using `"!import1/private"`.
```

Comment:
```suggestion
In this example, `"!import1/private/*"` is not reincluding the modules inside `private` because the negation mark (`!`) does not reinclude the files if it's parent directory is excluded by a pattern. In this case, `import1/private` directory is already excluded by the `import1/*` pattern. (The excluded directory can be reincluded using `"!import1/private"`.)
```